### PR TITLE
Self un-enroll LTI (1.3) tool

### PR DESCRIPTION
**Pull request by cmurtaugh**
This PR adds a new LTI 1.3 tool that can be installed into a course in order to let enrollees remove themselves from the course.

The tool appears in the right-hand navigation on the course homepage: "Un-enroll from this course". When a user clicks on the link, we check to make sure that self-enrollment is enabled for this course and that the user is actually self-enrolled. If so, the user is presented with a "Leave this course" button.

When the user clicks the button, the tool removes each enrollment with the source "selfenroll" for this user in this course. It also calls the Canvas API to remove each self-enrolled role from the Canvas course. 

See the README for instructions on the LTI 1.3 installation process. This PR includes three new management commands to facilitate the installation process: `generate_lti_key`, `create_tool_config`, and `list_lti_keys`. 

This PR also modifies the `self_enrollment_tool` so that it will install the new unenrollment LTI tool into the Canvas site when an administrator enables self-enrollment for that course.

### CODE REVIEW

**Review by haydn9000 — APPROVED** [review]
This looks good 👍 I noticed GitHub is warning about not having new lines at the end of the template files though. Not sure how much that matters

File: cross_list_courses/templates/add_new.html [hunk 1]
```diff
@@ -138,7 +138,7 @@ <h3>
             "bInfo" : false,
             "aoColumnDefs": [ { "orderable": false, "targets": [1,3] } ],
             "oLanguage": {
-              "sSearch": "Filter the list below"
+              "sSearch": "Filter the list"
```
Comment (sapnamysore) [inline cross_list_courses/templates/add_new.html:141]: Was. this text change intended for the cross listing tool ? Or is it a search and replace by-product while fixing the self reg tool?

File: cross_list_courses/templates/list.html [hunk 2]
```diff
@@ -120,7 +120,7 @@ <h4 class="modal-title bottom-spacing">Cross-Listed Courses in Current Terms:</h
             "bInfo" : false,
             "aoColumnDefs": [ { "orderable": false, "targets": 3 } ],
             "oLanguage": {
-              "sSearch": "Filter the list below"
+              "sSearch": "Filter the list"
```
Comment (sapnamysore) [inline cross_list_courses/templates/list.html:123]: Same as above- was this change intended for cross listing tool?

File: cross_list_courses/templates/add_new.html (hunk 1 above)
Comment (cmurtaugh) [reply cross_list_courses/templates/add_new.html:141]: Whoops -- an overly aggressive search-and-replace! :-)

File: cross_list_courses/templates/list.html (hunk 2 above)
Comment (cmurtaugh) [reply cross_list_courses/templates/list.html:123]: Ditto - I'll back these out

File: self_unenrollment_tool/views.py [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+def launch(request: HttpRequest):
+    message_launch = get_message_launch(request)
+    # this next line is necessary even if the variable is not used; if get_launch_data() is not called, launch data will not be saved to the session!
+    launch_data = message_launch.get_launch_data()
+    logger.info('self_unenrollment_tool launch', extra={'launch_data': launch_data})
+    return redirect(reverse('self_unenrollment_tool:index', kwargs={'launch_id': message_launch.get_launch_id()}))
+
+
+@require_lti_launch
+def index(request: HttpRequest, launch_id):
```
Comment (PatrickMTonne) [inline self_unenrollment_tool/views.py:52]: Nice use of type hints. Would it be worth adding types to the other arguments in each view for consistency?

File: canvas_account_admin_tools/settings/base.py [hunk 4]
```diff
@@ -326,7 +330,7 @@
     'loggers': {
         'gunicorn': {
             'handlers': ['gunicorn', 'console'],
-            'level': 'INFO',
+            'level': logging.WARNING,
```
Comment (PatrickMTonne) [inline canvas_account_admin_tools/settings/base.py:333]: I assume this is to reduce the noisiness of the `gunicorn` logs?

File: self_enrollment_tool/templates/self_enrollment_tool/add_new.html [hunk 5]
```diff
@@ -6,7 +6,7 @@
   <h3>
     <a href="{% url 'dashboard_account' %}">Admin Console</a>
     <small><i class="fa fa-chevron-right"></i></small>
-      <a href="{% url 'self_enrollment_tool:index' %}">Enable Self Enrollment</a>
+      <a href="{% url 'self_enrollment_tool:index' %}">Self Enrollment</a>
```
Comment (PatrickMTonne) [inline self_enrollment_tool/templates/self_enrollment_tool/add_new.html:9]: 👍

**Comment by sapnamysore** [comment]
Changes looked good to me👍  . The code walkthru was helpful to understand the logic, workflow and the LTI 1.3 details.. Nice work!

**Review by PatrickMTonne — APPROVED** [review]
👍 Looks good! Thanks for walking us through this earlier.

File: self_unenrollment_tool/views.py (hunk 3 above)
Comment (cmurtaugh) [reply self_unenrollment_tool/views.py]: :+1:

File: canvas_account_admin_tools/settings/base.py (hunk 4 above)
Comment (cmurtaugh) [reply canvas_account_admin_tools/settings/base.py:333]: Yeah -- we send tons of gunicorn log events to Splunk but they're not that useful. I think we really only need to know about warnings and above.
